Assets/Scripts/Game/WeaponManager.cs

```diff
@@ -228,25 +230,40 @@ void Update()
                 return;
             }
 
+            // Do nothing if player paralyzed or is climbing
+            if (GameManager.Instance.PlayerEntity.IsParalyzed || GameManager.Instance.ClimbingMotor.IsClimbing)
+            {
+                ShowWeapons(false);
+                return;
+            }
+
+            bool doToggleSheath = false;
+
             // Hide weapons and do nothing if spell is ready or cast animation in progress
             if (GameManager.Instance.PlayerEffectManager)
             {
                 if (GameManager.Instance.PlayerEffectManager.HasReadySpell || GameManager.Instance.PlayerSpellCasting.IsPlayingAnim)
                 {
-                    ShowWeapons(false);
-                    return;
-                }
-            }
+                    if (!isAttacking && InputManager.Instance.ActionStarted(InputManager.Actions.ReadyWeapon) && entityEffectManager)
```

Comment:
You can use GameManager.Instance.PlayerEffectManager here rather than grab another reference in Start.

Reply:
Updated!